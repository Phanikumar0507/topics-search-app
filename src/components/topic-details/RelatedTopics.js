import React from "react";
import Topic from "./Topic";

const RelatedTopics = ({ repoNode, setShowRelated }) => {
  return (
    <div className="modal-popup">
      <div className="modal-content">
        <div className="modal-header">
          <h3>Related Topics</h3>
          <span className="close-btn" onClick={() => setShowRelated(false)}>
            &times;
          </span>
        </div>
        <div className="topics-list">
          {repoNode.topic.relatedTopics.map((relatedTopic, index) => (
            <Topic key={index} topic={relatedTopic} setShowRelated={() => {}} />
          ))}
          {repoNode.topic.relatedTopics.length === 0 && (
            <div className="no-data-text">No related topics found.</div>
          )}
        </div>
      </div>
    </div>
  );
};

export default RelatedTopics;
