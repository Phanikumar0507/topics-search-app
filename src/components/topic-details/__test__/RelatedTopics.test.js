import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import RelatedTopics from "../RelatedTopics";

describe("Related Topics Component", () => {
  it("should render wihtout crash", () => {
    const repoNode = {
      topic: { relatedTopics: [{ name: "angular", stargazerCount: 10 }] },
    };
    const mockFun = jest.fn();
    const component = shallow(
      <RelatedTopics repoNode={repoNode} setShowRelated={mockFun} />
    );
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("should render no related topic found text on empty related topics", () => {
    const repoNode = {
      topic: { relatedTopics: [] },
    };
    const mockFun = jest.fn();
    const component = shallow(
      <RelatedTopics repoNode={repoNode} setShowRelated={mockFun} />
    );

    const divNode = component.find(".no-data-text");
    expect(divNode.text()).toBe("No related topics found.");
  });

  it("should call the setShowRelated method with false value", () => {
    const repoNode = {
      topic: { relatedTopics: [] },
    };
    const mockFun = jest.fn();
    const component = shallow(
      <RelatedTopics repoNode={repoNode} setShowRelated={mockFun} />
    );
    component.find(".close-btn").simulate("click");
    expect(mockFun).toHaveBeenCalledTimes(1);
    expect(mockFun).toHaveBeenCalledWith(false);
  });
});
