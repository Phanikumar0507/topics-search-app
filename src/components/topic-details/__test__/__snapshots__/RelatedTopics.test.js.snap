// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Topics Component should render wihtout crash 1`] = `
<div
  className="modal-popup"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <h3>
        Related Topics
      </h3>
      <span
        className="close-btn"
        onClick={[Function]}
      >
        Ã—
      </span>
    </div>
    <div
      className="topics-list"
    >
      <Topic
        key="0"
        setShowRelated={[Function]}
        topic={
          Object {
            "name": "angular",
            "stargazerCount": 10,
          }
        }
      />
    </div>
  </div>
</div>
`;
